zerver/views/message_edit.py

```diff
@@ -48,18 +50,20 @@ def fill_edit_history_entries(message_history: List[Dict[str, Any]], message: Me
                message_history[0]['timestamp'])
 
     for entry in message_history:
-        entry['topic'] = prev_topic
-        if LEGACY_PREV_TOPIC in entry:
+        filled_entry = entry.copy()
```

Comment:
It might be better to make the output struct here a `dataclass` and then explicitly initialize it from the properties of `entry`, just for readability. (Or even make both dataclasses, `CompressedMessageEditHistoryEntry` and `MessageEditHistoryEntry`, or whatever).

Reply:
I prefer the explicit method; the reason I did it this way was [the discussion starting with your message here](https://chat.zulip.org/#narrow/stream/3-backend/topic/edit.20history/near/901543), and specifically your confirmation about ten messages down of [my message here](https://chat.zulip.org/#narrow/stream/3-backend/topic/edit.20history/near/902087).

A bit further down, [here](https://chat.zulip.org/#narrow/stream/3-backend/topic/edit.20history/near/902180), you mentioned `TypedDict`, and in response to Steve mentioning `dataclass`, you said:

> Well, the point of this is to be marshalled as a dict on the wire, so I think TypedDict might be more appropriate?

I actually have a stashed patch that looks nearly finished for a `TypedDict` implementation of both types in `do_update_message()` and `fill_edit_history_entries()`, which I was going to integrate soon, but it wouldn't be difficult to convert it to `dataclass` if that's the way you want to go.

Reply:
I don't have strong feelings on the difference between the two; @andersk may have more thoughts.